dashboard/src/actions/kube.tsx

```diff
@@ -0,0 +1,39 @@
+import { ThunkAction } from "redux-thunk";
+import { ActionType, createAction } from "typesafe-actions";
+import { Kube } from "../shared/Kube";
+import { IResource, IStoreState } from "../shared/types";
+
+export const requestResource = createAction("REQUEST_RESOURCE", resolve => {
+  return (resourceID: string) => resolve(resourceID);
+});
+
+export const receiveResource = createAction("RECEIVE_RESOURCE", resolve => {
+  return (resource: { [resourceID: string]: IResource }) => resolve(resource);
+});
+
+export const receiveResourceError = createAction("RECEIVE_RESOURCE_ERROR", resolve => {
+  return (resource: { [resourceID: string]: Error }) => resolve(resource);
+});
+
+const allActions = [requestResource, receiveResource, receiveResourceError];
+
+export type KubeAction = ActionType<typeof allActions[number]>;
+
+export function getResource(
+  apiVersion: string,
+  resource: string,
+  namespace: string,
+  name: string,
+  query?: string,
+): ThunkAction<Promise<void>, IStoreState, null, KubeAction> {
+  return async dispatch => {
+    const key = Kube.getResourceURL(apiVersion, resource, namespace, name, query);
+    dispatch(requestResource(key));
+    try {
+      const r = await Kube.getResource(apiVersion, resource, namespace, name, query);
+      dispatch(receiveResource({ [key]: r }));
```

Comment:
In order to simplify all the transformation done here and in the reducer, wouldn't make more sense if you send `{ Key: key, resource: r }`

That will simplify the reducer that now does some transformation based on the position of the args and so on. Or there is a reason you want to send them as a map?

Reply:
:+1: I agree, I think that would be more clear.

Reply:
👍